enzyme/Enzyme/CApi.h

```diff
@@ -117,13 +117,42 @@ typedef enum {
   DEM_ForwardModeSplit = 4,
 } CDerivativeMode;
 
+/// Generates a new function based on the input, which uses forward-mode AD.
+///
+/// @param mode: DEM_ForwardMode and DEM_ForwardModeSplit
+/// @param retType: DFT_DUP_ARG, DFT_CONSTANT
+/// @param constant_args: pointing to combinations of DFT_DUP_ARG and
+/// DFT_CONSTANT
+/// @param width: integer n >= 1. The generated functions expects n additional
+/// inputs for each input marked as DFT_DUP_ARG.
+/// @param returnValue: Whether to return the primary return value.
+/// The similar dret_used is not available here / implicitely set to true, as it
+/// is the only place where we return gradient information.
+/// DEM_ReverseModeCombined. Otherwise, should be set to the amount of
+/// input params of `todiff`, which might change between the forward and the
+/// reverse pass.
 LLVMValueRef EnzymeCreateForwardDiff(
     EnzymeLogicRef, LLVMValueRef todiff, CDIFFE_TYPE retType,
     CDIFFE_TYPE *constant_args, size_t constant_args_size,
     EnzymeTypeAnalysisRef TA, uint8_t returnValue, CDerivativeMode mode,
     unsigned width, LLVMTypeRef additionalArg, struct CFnTypeInfo typeInfo,
     uint8_t *_uncacheable_args, size_t uncacheable_args_size);
 
+/// Generates a new function based on the input, which uses reverse-mode AD.
+///
+/// Based on the
+/// @param retType: When returning f32/f64 we might use DFT_OUT_DIFF.
```

Comment:
I would use llvm type names here: float/double. We also would want to state the general rule that out_diff is used when the value being differentiated is being passed as a register (examples being float double, a vector of these, an llvm.struct of these, etc) rather than through memory (e.g. a pointer)